### DISCUSSION TITLE
API versions

### DISCUSSION THREAD

**Discussion by mcsiki** [Azure/CanadaPubSecALZ#369] [General]
Hi Community,

I realized that most resources use outdated and/or preview API versions. For example:
- Microsoft.Authorization/roleAssignments@2020-04-01-preview
- Microsoft.Network/virtualNetworks@2021-02-01
- Microsoft.KeyVault/vaults@2021-04-01-preview

My questions:
Is it safe to use the most up-to-date versions for my deployments? I assume yes.
Does someone, some group is doing version updates?
What is the regular cadence for updating API version?

Thanks

**Comment by SenthuranSivananthan** [comment]
@mcsiki In most cases, it's safe to upgrade the API version to the latest.  Each API version is managed the respective service team (e.g. Key vault team managed the key vault APIs, Network team manages the network APIs, etc.) so the cadence varies and it is based on the changes that are made to the underlying Azure Management REST APIs.

When upgrading to the new API version, you will get new features and default settings so you'll want to review your environment to make sure it does not conflict with your org's controls.
